server/src/test/java/org/elasticsearch/cluster/routing/allocation/allocator/AllocationBalancingRoundSummaryServiceTests.java

```diff
@@ -0,0 +1,251 @@
+import org.elasticsearch.core.TimeValue;
+import org.elasticsearch.test.ESTestCase;
+import org.elasticsearch.test.MockLog;
+import org.elasticsearch.threadpool.ThreadPool;
+import org.junit.Before;
+
+public class AllocationBalancingRoundSummaryServiceTests extends ESTestCase {
+    private static final Logger logger = LogManager.getLogger(AllocationBalancingRoundSummaryServiceTests.class);
+
+    final Settings enabledSummariesSettings = Settings.builder()
+        .put(AllocationBalancingRoundSummaryService.ENABLE_BALANCER_ROUND_SUMMARIES_SETTING.getKey(), true)
+        .build();
+    final Settings disabledDefaultEmptySettings = Settings.builder().build();
+    final Settings enabledButNegativeIntervalSettings = Settings.builder()
+        .put(AllocationBalancingRoundSummaryService.ENABLE_BALANCER_ROUND_SUMMARIES_SETTING.getKey(), true)
+        .put(AllocationBalancingRoundSummaryService.BALANCER_ROUND_SUMMARIES_LOG_INTERVAL_SETTING.getKey(), TimeValue.MINUS_ONE)
+        .build();
+
+    ClusterSettings enabledClusterSettings = new ClusterSettings(enabledSummariesSettings, ClusterSettings.BUILT_IN_CLUSTER_SETTINGS);
+    ClusterSettings disabledDefaultEmptyClusterSettings = new ClusterSettings(
+        disabledDefaultEmptySettings,
+        ClusterSettings.BUILT_IN_CLUSTER_SETTINGS
+    );
+    ClusterSettings enabledButNegativeIntervalClusterSettings = new ClusterSettings(
+        enabledButNegativeIntervalSettings,
+        ClusterSettings.BUILT_IN_CLUSTER_SETTINGS
+    );
+
+    // Construction parameters for the service.
+
+    DeterministicTaskQueue deterministicTaskQueue;
+    ThreadPool testThreadPool;
+
+    @Before
+    public void setUpThreadPool() {
+        deterministicTaskQueue = new DeterministicTaskQueue();
+        testThreadPool = deterministicTaskQueue.getThreadPool();
+    }
+
+    /**
+     * Test that the service is disabled and no logging occurs when
+     * {@link AllocationBalancingRoundSummaryService#ENABLE_BALANCER_ROUND_SUMMARIES_SETTING} defaults to false.
+     */
+    public void testServiceDisabledByDefault() {
+        var service = new AllocationBalancingRoundSummaryService(testThreadPool, disabledDefaultEmptyClusterSettings);
+
+        try (var mockLog = MockLog.capture(AllocationBalancingRoundSummaryService.class)) {
+            /**
+             * Add a summary and check it is not logged.
+             */
+
+            service.addBalancerRoundSummary(new BalancingRoundSummary(50));
+            service.verifyNumberOfSummaries(0); // when summaries are disabled, summaries are not retained when added.
+            mockLog.addExpectation(
+                new MockLog.UnseenEventExpectation(
+                    "Running balancer summary logging",
+                    AllocationBalancingRoundSummaryService.class.getName(),
+                    Level.INFO,
+                    "Balancing round summaries:*"
+                )
+            );
```

Comment:
Nit: I wonder if this would be more robust if it just matched anything from `AllocationBalancingRoundSummaryService`, because we're asserting that we DON'T see something I think it's better to not match a specific message if possible? that way if the message changes and this goes unnoticed it'll still be effective.

Reply:
I've changed the Unseen to "*" matches, and Seen to a constant to avoid duplicate hardcoding.